### PR TITLE
Added ONNX model checker to ONNX export

### PR DESCRIPTION
**Pull request by bpstark**
Included the ONNX model checker code in the ONNX export
this will force onnx checker to run for all models that get exported.
This should help with validating exported models.

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 30038918:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=30038918547ba2412998cc1d8982779b2928ecbc).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4274428) pytorch_python_doc_push (1/1)
**Step:** "Doc Build and Push" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87200841) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87200841)) :snowflake:
<details>
<summary>
<code>$<3>$<3> c5f14fd98ba9: Pushing   2.49GB/6.02GB $<3>$<3> c5f14fd98ba9: Pushing   6.02GB/6.02GB toomanyrequests: Rate exceeded</code>
</summary>

```
$<3>$<3> eea1661bdba1: Preparing   
$<3>$<3> 5bb9fbc1099e: Preparing   
$<3>$<3> 0d9b5dad09df: Preparing   
$<3>$<3> a80eb84c4169: Preparing   
$<3>$<3> 85775ce071b3: Preparing   
$<3>$<3> bfd07f554dc2: Preparing   
$<3>$<3> 00c6fc364d3a: Preparing   
$<3>$<3> 77008e118980: Preparing   
$<3>$<3> 6cb741cb00b7: Preparing   
$<3>$<3> f36b28e4310d: Preparing   
c5f14fd98ba9: Pushing   2.49GB/6.02GB $<3>$<3> c5f14fd98ba9: Pushing   6.02GB/6.02GB toomanyrequests: Rate exceeded 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -2454,6 +2454,27 @@ def forward(self, input):
         x = torch.randn(1, 2, 3, requires_grad=True)
         self.run_test(EmptyBranchModel(), x)
 
+    def test_onnx_proto_checker(self):
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: Let's increase the variety of test coverage here. We want to ensure that failures are indeed captured, and warnings emitted.
Checking for invalid opset version, invalid operator definition, and SSA format are good examples.
Also do we know if the export is going to fail for models with custom operators?

**Review by BowenBao — CHANGES_REQUESTED** [review]
This is a good start to improve our test coverages for ONNX exporter utilizing existing tools. Some inline comments regarding test coverages.

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Comment by bpstark** [comment]
@houseroad can you please take a look at this.
Thanks

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -485,6 +485,10 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
                 {}, opset_version, dynamic_axes, False, operator_export_type,
                 strip_doc_string, val_keep_init_as_ip, custom_opsets, val_add_node_names)
 
+        if operator_export_type != OperatorExportTypes.ONNX_ATEN_FALLBACK:
```
Comment (houseroad) [inline torch/onnx/utils.py]: Shall we add a parameter in torch.onnx.export to control this check? I am wondering if someone would like to export some special (also invalid) model, then we suddenly block them.

Comment (bpstark) [reply torch/onnx/utils.py]: yeah thats probably a good idea to be able to disable it if needed.

**Comment by houseroad** [comment]
Please fix the conflict and let the test pass

**Comment by bpstark** [comment]
@houseroad sorry not sure what conflict you are referring to

**Comment by houseroad** [comment]
nvm, seems it's okay now. could you update the doc here as well? https://github.com/pytorch/pytorch/blob/9a23e2ff0602298ebc0591c70fa92ddac6facd2b/torch/onnx/__init__.py

**Comment by bpstark** [comment]
@houseroad can you please pull the latest changes now that I have updated as requested.

Thanks

**Review by houseroad — APPROVED** [review]
Thanks!

**Pull request closed without merging**